src/app/components/Loop/index.tsx

```diff
@@ -0,0 +1,425 @@
+      o =>
+        o.status === 'OPEN' &&
+        parseInt(o.local_balance, 10) > parseInt(loopTerms.min_swap_amount, 10),
+    );
+    // Only return channels with enough loot for looping in
+    const openChannelsLoopIn = channels.filter(
+      o =>
+        o.status === 'OPEN' &&
+        parseInt(o.capacity, 10) - parseInt(o.local_balance, 10) >
+          parseInt(loopTerms.min_swap_amount, 10),
+    );
+
+    // Get channels to choose from
+    const loopOutItems = openChannelsLoopOut.map(c => (
+      <Menu.Item
+        key={c.channel_point}
+        onClick={() => this.handleSetChannelId(c.channel_point, openChannelsLoopOut)}
+      >
+        {`${c.node.alias} => ${c.local_balance} sats available`}
+      </Menu.Item>
+    ));
+    const loopInItems = openChannelsLoopIn.map(c => (
+      <Menu.Item
+        key={c.channel_point}
+        onClick={() => this.handleSetChannelId(c.channel_point, openChannelsLoopIn)}
+      >
+        {`${c.node.alias} => ${c.local_balance} sats available`}
+      </Menu.Item>
+    ));
+
+    const loopInMenu = <Menu>{loopInItems}</Menu>;
+    const loopOutMenu = <Menu>{loopOutItems}</Menu>;
+
+    // Destructure the state
+    const {
+      isAnyValue,
+      amount,
+      loopType,
+      destination,
+      swapRoutingFee,
+      swapFee,
+      minerFee,
+      prepayAmt,
+      channel,
+      quoteModalIsOpen,
+      advanced,
+      conf,
+      htlc,
+    } = this.state;
+    const actions: ButtonProps[] = [
+      {
+        children: (
+          <>
+            <Icon type="question-circle" theme="filled" /> {`${loopType} Quote`}
+          </>
+        ),
+        type: 'primary' as any,
+      },
+    ];
+
+    if (loopTerms === null) {
+      return null;
+    }
+
+    return (
+      <>
+        <div>
+          <Radio.Group defaultValue="a" size="large">
+            <Radio.Button value="a" onClick={this.setLoopOutType}>
+              Loop Out
+            </Radio.Button>
+            <Radio.Button value="b" onClick={this.setLoopInType}>
+              Loop In
+            </Radio.Button>
+          </Radio.Group>
+        </div>
+        <div className="Loop">
+          <InputLoopAddress
+            setLoop={this.props.setLoop}
+            setLoopIn={this.props.setLoopIn}
+            error={error}
+            initialUrl={this.props.url}
+            type={loopType}
+          />
+          {loopTerms.swap_fee_base !== '' && (
+            <Tooltip
+              overlayClassName="Loop-terms-tip"
+              title={`
+                Base fee: ${loopTerms.swap_fee_base} sats |
+                Fee rate: ${loopTerms.swap_fee_rate} sats |
+                Prepay amt: ${
+                  loopTerms.prepay_amt === undefined ? '1337' : loopTerms.prepay_amt
+                }
+                Min Swap amt: ${loopTerms.min_swap_amount} sats |
+                Max Swap amt: ${loopTerms.max_swap_amount} sats |
+                CLTV delta: ${loopTerms.cltv_delta}
+              `}
```

Comment:
This tooltip displays way too much data to work well as a tooltip. It should either be persistently rendered inline, or toggle / collapsable.

Reply:
Replaced tooltip with collapse